File: data/cgi/CGI.pl
#!/usr/bin/perl

use strict;
use warnings;
use CGI qw(:standard);

print header(-type => 'text/html');
print start_html(-title => 'CGI Response Perl',
                 -style => {'src' => 'https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css'},
                 -head  => meta({'charset' => 'UTF-8'}));
print '<nav class="navbar navbar-expand navbar-dark bg-dark">';
print '<a class="navbar-brand" href="#">webserv42</a>';
print '<ul class="navbar-nav">';
print '<li class="nav-item">';
print '<a class="nav-link" href="../index.html">Accueil</a>';
print '</li>';
print '<li class="nav-item">';
print '<a class="nav-link" href="../televerser.html">Téléverser</a>';
print '</li>';
print '<li class="nav-item">';
print '<a class="nav-link" href="../team.html">Notre équipe</a>';
print '</li>';
print '<li class="nav-item">';
print '<a class="nav-link" href="../cgi.html">CGI</a>';
print '</li>';
print '</ul>';
print '</nav>';
print start_body();
print "<h1>Test CGI with Perl</h1>";
print "<p>Hello World!</p>";

my $query = CGI->new;

if (defined $query->param('name')) {
    print "<p>Your name is: " . $query->param('name') . "</p>";
}

if (defined $query->param('login')) {
    print "<p>Your login is: " . $query->param('login') . "</p>";
}

print "<h3>Environments variables:</h3>";
foreach my $key (sort keys %ENV) {
    print "<li>$key: $ENV{$key}</li>";
}

print end_html();

File: data/cgi/CGI.php
#!/usr/bin/php-cgi
<?php
// Récupération des données du formulaire
$name = isset($_POST['name']) ? $_POST['name'] : '';
$login = isset($_POST['login']) ? $_POST['login'] : '';

// Envoi de l'entête HTTP
header('Content-Type: text/html; charset=utf-8');

// Affichage de la page HTML
echo <<<EOT
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>CGI Response PHP</title>
    <style>
        .navbar .active > a {
            background-color: #343a40;
            color: white;
        }
    </style>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" crossorigin="anonymous">
</head>
<body>
<nav class="navbar navbar-expand navbar-dark bg-dark">
    <a class="navbar-brand" href="#">webserv42</a>
    <ul class="navbar-nav">
        <li class="nav-item">
            <a class="nav-link" href="../index.html">Accueil</a>
        </li>
        <li class="nav-item">
            <a class="nav-link" href="../televerser.html">Téléverser</a>
        </li>
        <li class="nav-item">
            <a class="nav-link" href="../team.html">Notre équipe</a>
        </li>
        <li class="nav-item">
            <a class="nav-link" href="../cgi.html">CGI</a>
        </li>
    </ul>
</nav>
<h1>Test CGI PHP</h1>
<p>Hello World!</p>
EOT;

// Affichage des données du formulaire
if (!empty($name)) {
    echo "<p>Your name is: $name</p>";
}
if (!empty($login)) {
    echo "<p>Your login is: $login</p>";
}

// Affichage des variables d'environnement
echo "<h3>Environments variables:</h3>";
foreach ($_SERVER as $key => $value) {
    echo "<li>$key: $value</li>";
}

echo '</body></html>';